src/peft/utils/pissa_utils.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Reference paper: https://arxiv.org/abs/2404.02948
+
+from __future__ import annotations
+
+
+import os
+import torch
+from safetensors import safe_open
+from safetensors.torch import save_file
+import json
+
+
+def pissa_pre_training_saving(peft_model, tokenizer, save_path, push_to_hub=None):
+    # No need for SVD when we load the PiSSA and residual model saved locally.
+    peft_model.peft_config["default"].init_lora_weights = True
+    # Save PiSSA adapter.
+    peft_model.save_pretrained(os.path.join(save_path, "pissa_init"))
+
+    # Directly delete default adapter with `peft_model.delete_adapter("default")` and save as base model with `peft_model.get_base_model()` will raise an error (At least one adapter is needed):
+    # File ~/peft/src/peft/peft_model.py:926, in PeftModel.active_peft_config(self)
+    #     925 def active_peft_config(self):
+    # --> 926     return self.peft_config[self.active_adapter]
+    # KeyError: 'default'
+
+    # Therefore, we merging the residual with a zero initialized lora, which will not influence its value.
+    peft_model.add_adapter(adapter_name="zero_initialized_lora", peft_config=peft_model.peft_config["default"])
+    peft_model.merge_and_unload(adapter_names=["zero_initialized_lora"])
+    model = peft_model.get_base_model()
```

Comment:
Instead of going through all these steps, can we not do `model = peft_model.unload()`?